internal/provider/resource_user.go

```diff
@@ -0,0 +1,179 @@
+package provider
+
+import (
+	"context"
+	"regexp"
+	"time"
+
+	"github.com/PacketFabric/terraform-provider-packetfabric/internal/packetfabric"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+)
+
+func resourceUser() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceUserCreate,
+		ReadContext:   resourceUserRead,
+		UpdateContext: resourceUserUpdate,
+		DeleteContext: resourceUserDelete,
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(10 * time.Minute),
+			Read:   schema.DefaultTimeout(10 * time.Minute),
+			Update: schema.DefaultTimeout(10 * time.Minute),
+			Delete: schema.DefaultTimeout(10 * time.Minute),
+		},
+		Schema: map[string]*schema.Schema{
+			"login": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: validation.StringLenBetween(1, 255),
+				Description:  "User login.",
+			},
+			"first_name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: validation.StringLenBetween(1, 255),
+				Description:  "User first name.",
+			},
+			"last_name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: validation.StringLenBetween(1, 255),
+				Description:  "User last name.",
+			},
+			"email": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validation.StringIsNotEmpty,
+				Description:  "User e-mail.",
+			},
+			"phone": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: validation.StringMatch(regexp.MustCompile(`^[0-9 ()+.-]+(\s?(x|ex|ext|ete|extn)?(\.|\.\s|\s)?[\d]{1,9})?$`), "Phone number must match the pattern ^[0-9 ()+.-]+(\\s?(x|ex|ext|ete|extn)?(\\.|\\.\\s|\\s)?[\\d]{1,9})?$"),
+				Description:  "User phone number.",
+			},
+			"password": {
+				Type:         schema.TypeString,
+				Required:     true,
+				Sensitive:    true,
+				ForceNew:     true,
```

Comment:
Changing the password requires creating a new user?

Reply:
This is what I found actually, on the user himself can change the password after it has been created. Same for the email, that's why I added the forcenew so the user is recreated if the admin want to change user's password.

Reply:
I found this API `/v2/users/reset-password`  so I will use it and also add warning/info around password update.